compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -4604,3 +4604,49 @@ declare_lint! {
         reference: "issue #X <https://github.com/rust-lang/rust/issues/X>",
     };
 }
+
+declare_lint! {
+    /// The `private_macro_use` lint detects macros that are imported
+    /// with `#[macro_use]`.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,ignore (needs extern crate)
+    /// // extern_macro.rs
+    /// macro_rules! foo { () => {}; }
```

Comment:
```suggestion
    /// use proc_macro_foo::foo;
```
Usual `macro_rules!` are not affected by this issue, only `macro`s 2.0, and reexports of procedural macros or `#[macro_export] macro_rules!` from some third crate.

For a macro to be fetched by `#[macro_use] extern crate bar;` its name should be placed into the `bar`'s root module, but `macro_rules!` names are not placed into modules.

Reply:
I will modify as follows:

```rs
// extern_macro.rs
macro_rules! foo_ { () => {}; }
use foo_ as foo;
```

Reply:
Ah, yes, that should work too.